src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -261,10 +272,13 @@ internal static void GetSolutionFileAndVisualStudioMajorVersions(string solution
             solutionVersion = 0;
             visualStudioMajorVersion = 0;
 
+            // Solutions can be passed as either a solution or a solution filter or truncated list of what projects to build.
+            string realSolutionFile = FileUtilities.IsSolutionFilterFilename(solutionFile) ? solutionFile.Substring(0, solutionFile.Length - 1) : solutionFile;
```

Comment:
Does this method need to be aware of slnf at all? Couldn't it just always be given a real solution?